### ISSUE TITLE
`@typescript-eslint/no-unnecessary-condition` false positive when property is defined anywhere other than the last line of `script` tag

### ISSUE THREAD

**Issue by DetachHead** [sveltejs/eslint-plugin-svelte3#191]
```svelte
<script>
    let foo = undefined
    1 // remove this line and the error goes away
</script>

<button on:click={() => (foo = 'asdf')}>asdf</button>
{#if foo !== undefined}
    <p>clicked</p>
{/if}
```
```
7:6  error  Unnecessary conditional, both sides of the expression are literal values  @typescript-eslint/no-unnecessary-condition

✖ 1 problem (1 error, 0 warnings)
```

**Comment by DetachHead** [comment]
this issue doesn't occur in [`eslint-plugin-svelte`](https://github.com/ota-meshi/eslint-plugin-svelte) which seems to be more actively maintained

- #184
